Adds Compatability for Conversations in 1.21.3 using own PacketWrapper for Steerpackets

DanielSchmerber opened the pull request:
<!-- Please describe your changes here. -->


---

### Related Issues

<!-- Issue number if existing. -->
Closes #XXXX

### Requirements
- [x] I made sure my contribution fulfills the [requirements](https://betonquest.org/DEV/Participate/Process/Submitting-Changes/#checklist).

### Reviewer's checklist

<!-- DON'T DO ANYTHING HERE -->
<!-- This is a checklist for the reviewers, and will be checked by them! -->
Did the contributor...
- [x]  ... test their changes?
- [x]  ... increment the [Version](https://betonquest.org/DEV/Participate/Misc/Versioning-and-Releasing/#versioning)?
- [x]  ... update the [Changelog](https://betonquest.org/DEV/Participate/Process/Maintaining-the-Changelog/)?
- [x]  ... update the [Documentation](https://betonquest.org/DEV/Participate/Process/Docs/Workflow/)?
- [x]  ... adjust the [ConfigPatcher](https://betonquest.org/DEV/API/ConfigurationFiles/#updating-configurationfiles)?
- [x]  ... clean the commit history?

Check if the build pipeline succeeded for this PR!

Wolf2323 commented on src/main/java/org/betonquest/betonquest/compatibility/protocollib/wrappers/WrapperPlayClientSteerVehicleUpdated.java:
```diff
@@ -0,0 +1,88 @@
+
+    }
+    @Override
+    public void setForward(final float value) {
+        throw new UnsupportedOperationException(NOT_IMPLEMENTED);
+    }
+    @Override
+    public boolean isJump() {
+        return this.handle.getStructures().read(0).getBooleans().read(4);
+    }
+    @Override
+    public void setJump(final boolean value) {
+        throw new UnsupportedOperationException(NOT_IMPLEMENTED);
+    }
+    @Override
+    public boolean isUnmount() {
+        return this.handle.getStructures().read(0).getBooleans().read(5);
+    }
+    @Override
+    public void setUnmount(final boolean value) {
+        throw new UnsupportedOperationException(NOT_IMPLEMENTED);
+    }
```
Why dont implement the missing methods, as its a real simple set? Or I am wrong?

DanielSchmerber replied:
Didnt implement them because they are not used

Some would be simple sets, the setForward  / setUnmount would require additional logic

Wolf2323 reviewed (changes requested):
There are some things you need to adjust, but over all it looks fine.
And you dont need to check the `Reviewer's checklist` thats for me ;)

TheosRee commented on CHANGELOG.md:
```diff
@@ -39,6 +39,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 ### Deprecated
 ### Removed
 ### Fixed
+- Conversations work in 1.21.3 again
```
New entries go at the bottom of the list

TheosRee commented on src/main/java/org/betonquest/betonquest/compatibility/protocollib/wrappers/WrapperPlayClientSteerVehicleUpdated.java:
```diff
@@ -0,0 +1,104 @@
+
+    }
+
+    /**
```
Here is the header JDoc missing

TheosRee reviewed (changes requested):
In general this looks great. The only thing is the spacing between characters which does not match the checkstyle rules and will fail the build.

DanielSchmerber commented:
> In general this looks great. The only thing is the spacing between characters which does not match the checkstyle rules and will fail the build.

I moved the changelog entry to the bottom, and updated the JDoc
can you tell me which spacing/stylesheet you are referring to?

DanielSchmerber commented:
> The build pipeline failed with some code style issues. You can see them here https://github.com/BetonQuest/BetonQuest/actions/runs/12059016993/job/33644246382?pr=3051 or by running `mvn verify` If you need help solving it, please take a look here: https://betonquest.org/DEV/Participate/Process/Code/Checking-Requirements/#fulfil-the-contributing-requirements and choose the checkstyle tab belog

Thanks for clarifiying it.
I adjusted the formatting according to the output of mvn verify and it now passes on my local pc

TheosRee commented:
@DanielSchmerber please squash your commits and rebase on current main
